test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
+      XDG_CONFIG_DIRS='siamese:persian',
+      XDG_DATA_HOME='dog',
+      XDG_DATA_DIRS='chihuahua:terrier',
+      XDG_CACHE_HOME='capybara',
+    }})
+  end)
+
+  it('knows the config home', function()
+    eq('cat/nvim', eval('stdpath("config")'))
+  end)
+
+  it('knows the data home', function()
+    if os_name() == 'windows' then
+      eq('dog\\nvim-data', eval('stdpath("data")'))
```

Comment:
I decided to leave the `os_name() == "windows"` in place for the case of appending `/nvim-data` (windows) vs. `/nvim` (everyone else) because I felt it was clearer.  I did move it into a `before_each` though to get it out of each test.